server/conn.go

```diff
@@ -1187,6 +1188,12 @@ func (cc *clientConn) dispatch(ctx context.Context, data []byte) error {
 	if cmd < mysql.ComEnd {
 		cc.ctx.SetCommandValue(cmd)
 	}
+	chs, err := variable.GetSessionOrGlobalSystemVar(vars, variable.CharacterSetResults)
+	if err != nil {
+		logutil.Logger(ctx).Info("get character_set_results system variable failed", zap.Error(err))
+	}
+	cc.textDumper = newTextDumper(chs)
```

Comment:
what it the query is multiple statement and one of them is `set @@character_set_results`?